### ISSUE TITLE
CAF Naming work

### ISSUE THREAD

**Issue by lawrencegripper** [Azure/terraform-azurerm-naming#13]
Not sure which came first, there is an existing provider published in Hashicorp registry (found by accident) which implements the [naming convensions from here](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/naming-and-tagging)

https://github.com/aztfmod/terraform-provider-azurecaf

**Comment by Nepomuceno** [comment]
Was not aware of that module previously. We started with the implementation of the conventions associated with that document but the number of resources that thar link offers it is really small compared to the ones that we have the objective of implementing with this provider. 

I like the idea of having that implemented as a provider given the limitation of modules including string manipulation Will definitely check that out but from what I found the approach it is quite different for both.
